src/indexCors.ts

```diff
@@ -0,0 +1,50 @@
+import cors from 'cors'
+import express, { Request, Response } from 'express'
+import fetch, {
+  RequestInfo,
+  RequestInit,
+  Response as FetchResponse
+} from 'node-fetch'
+
+const app = express()
+
+app.use(express.json())
+app.use(cors())
+
+app.all('*', async (req: Request, res: Response) => {
+  const proxyDomain = req.headers.proxydomain
+  delete req.headers.proxydomain
```

Comment:
Maybe instead of `proxydomain` we use `Host`? Typically, the client sends the Host header as the destination (even for proxies), but Idk if the `fetch` API gives us this control (testing needed). But if so, it'd make more sense to use the standard header fields.

```js
fetch('http://cors-proxy.edge.app', {
  headers: {
    'Host': 'destinationserver.com'
  }
});
```